module/VuFind/src/VuFind/View/Helper/Root/Content.php

```diff
@@ -0,0 +1,100 @@
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+use VuFind\Content\PageLocator;
+
+/**
+ * Content View Helper to resolve translated pages.
+ * This is basically a wrapper around the PageLocator.
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Content extends AbstractHelper
+{
+    /**
+     * PageLocator instance to resolve translated pages.
+     *
+     * @var PageLocator
+     */
+    protected $pageLocator;
+
+    /**
+     * Context View Helper instance to resolve translated pages.
+     *
+     * @var Context
+     */
+    protected $contextHelper;
+
+    /**
+     * Constructor
+     *
+     * @param PageLocator $pageLocator   Page locator
+     * @param Context     $contextHelper Context view helper
+     */
+    public function __construct(PageLocator $pageLocator,
+        Context $contextHelper
+    ) {
+        $this->pageLocator = $pageLocator;
+        $this->contextHelper = $contextHelper;
+    }
+
+    /**
+     * Search for a translated template and render it using a temporary context.
+     *
+     * @param string $pageName    Name of the page
+     * @param string $pathPrefix  Path where the template should be located
+     * @param array  $context     Optional array of context variables
+     * @param array  $pageDetails Optional output variable for additional info
+     * @param string $pattern     Optional file system pattern to search page
+     *
+     * @return string            Rendered template output
+     */
+    public function renderTranslated($pageName, $pathPrefix='content',
+        $context=[], &$pageDetails=[], $pattern=null
+    ) {
+        if (!str_ends_with($pathPrefix, '/')) {
+            $pathPrefix .= '/';
+        }
+        $pathPrefix = 'templates/' . $pathPrefix;
+        $pageDetails = $this->pageLocator->determineTemplateAndRenderer(
+            $pathPrefix, $pageName, $pattern
+        );
+        $path = preg_replace(
+            '"^.+' . $pageDetails['theme'] . '/templates/"', '',
```

Comment:
I think my original comment got lost in the refactoring, but I'm still concerned that this may not be entirely reliable, if VuFind is installed in certain weird directory paths. Would it make sense to replace `^.+` with the `APPLICATION_PATH` constant and do a straightforward `str_replace` instead of `preg_replace`? That might serve the same purpose but be much more exact.